python/pyarrow/_flight.pyx

```diff
@@ -37,6 +38,31 @@ import pyarrow.lib as lib
 cdef CFlightCallOptions DEFAULT_CALL_OPTIONS
 
 
+cdef int check_flight_status(const CStatus& status) nogil except -1:
+    cdef shared_ptr[FlightStatusDetail] detail
+
+    if status.ok():
+        return 0
+
+    if FlightStatusDetail.UnwrapStatus(status, &detail):
+        with gil:
+            message = frombytes(status.message())
+        if detail.get().code() == CFlightStatusInternal:
+            raise FlightInternalError(message)
```

Comment:
I _think_ all the raises should be under the `with gil`, though perhaps Cython does it for us?

Reply:
It "worked" before but I'd rather be explicit, thanks.